### ISSUE TITLE
User reports services are restarted when lock override is disabled

### ISSUE THREAD

**Issue by pipex** [balena-io/balena-supervisor#1563]
Steps to reproduce 

1. Stop all the running services 
2. Enable the lock overide 
3. Update is applied
4. Services restart
5. DISABLE the lock overide 
Services restart, again.

**Comment by 20k-ultra** [comment]
Those start/stop/restart actions are just a proxy to API endpoints that start/stop/restart a service. These events do not persist in the target state and that is by design.

So, the reason the application starts again (restarts) after enabling lock override is because enabling lock override does change the target state. This makes the cloud tell the device to check for latest target state of which it find `"RESIN_OVERRIDE_LOCK": "1",` and tries to apply. While applying the Supervisor notices that it has some stopped containers but the target state wants running containers so it then adds starting those containers to the list of steps to perform when trying to reach the target state.

Any kind of target state change is conveyed to the Supervisor by just saying "hey check for an update" not what the update is. So any differences will cause steps to be generated.

The solution to this is make the start/stop/restart actions affect the target state but I've discussed this before and that's not what we want. Might be worth bringing up again.

**Comment by pipex** [comment]
Not sure if that is the issue here @20k-ultra, if the service is already running, changing the lock override state will push a new target state, but it should not restart the service. 

One thing we should clarify with @joehounsham is what the step 3 means. `Update is applied` means that there was a pending update and it takes place once the lock override state changes right?

**Comment by joehounsham** [comment]
@pipex Yes that is right

**Comment by joehounsham** [comment]
@20k-ultra Did you mean to say above that  the application starts again (restarts) after **disabling** lock override is because disabling lock override does change the target state.?

**Comment by 20k-ultra** [comment]
@joehounsham re-reading this I might have misunderstood the sequence of events in the original comment from @pipex.

I was saying the following scenario is true; if you stopped a container and disabled lock override then that stopped container would be started. However, the scenario is actually with containers that are running, after you disable lock override they are restarted. I would not expect that to happen.

**Comment by joehounsham** [comment]
Yep, 'after you disable lock override they are restarted. I would not expect that to happen.' - we are on the same page now @20k-ultra
